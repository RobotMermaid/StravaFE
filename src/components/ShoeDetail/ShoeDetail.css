.section {
  margin: 1rem 1rem 1rem 1rem;
}