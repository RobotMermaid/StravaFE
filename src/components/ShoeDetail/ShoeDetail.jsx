import React  from 'react';
import PropTypes from 'prop-types';
import findShoeImg from '../Services/shoeImg';

const ShoeDetail = ({ model, terrain, durable_miles, avg_stars, shoes_id, img_src }) =>  {
  let img = findShoeImg(img_src);

  return (
  <section>
    <h3>{model}</h3>
    <img src={img} />
    <p>Best for running on {terrain}</p>
    <p>Rating: {avg_stars} stars</p>
    <p>Durable for up to {durable_miles} miles</p>
  </section>
)};


ShoeDetail.propTypes = {
  model: PropTypes.string,
  id: PropTypes.number,
  avg_stars: PropTypes.number,
  terrain: PropTypes.string,
  durable_miles: PropTypes.number,
};

export default ShoeDetail;