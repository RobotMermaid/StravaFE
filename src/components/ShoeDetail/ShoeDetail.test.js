import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ShoeDetail from './ShoeDetail';

describe('Shoe Detail component', () => {
  afterEach(() => cleanup());
  it('renders Shoe Detail', () => {
    let shoe = {
      id: 1,
      model: 'Altra', 
      terrain: 'trail', 
      durable_miles: 400, 
      avg_stars: 4.5, 
      shoes_id: 3
    }
    const { asFragment } = render(<ShoeDetail shoe={shoe}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});