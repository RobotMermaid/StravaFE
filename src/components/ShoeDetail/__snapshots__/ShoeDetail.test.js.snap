// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoe Detail component renders Shoe Detail 1`] = `
<DocumentFragment>
  <section>
    <h3 />
    <p>
      Best for: 
    </p>
    <p>
      Rating:  stars
    </p>
    <p>
      Durable for up to  miles
    </p>
  </section>
</DocumentFragment>
`;
